### DISCUSSION TITLE
Problem and Solution: `permission denied while trying to connect to the Docker daemon socket`

### DISCUSSION THREAD

**Discussion by haxys** [automatic-ripping-machine/automatic-ripping-machine#614] [General]
After installing docker [per the DigitalOcean guide](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-22-04) I began following the [ARM docker installation intructions](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/docker). When launching the `docker-setup.sh` script, everything ran fine until the script attempted to download the ARM docker image. Since I had already installed docker, the script's docker installation was skipped, and the new `arm` user was not added to the `docker` group, resulting in an error:

```
$ sudo ./docker-setup.sh 
Hit:1 https://downloads.plex.tv/repo/deb public InRelease
[...]
Docker installation detected, skipping...
Pulling image from automaticrippingmachine/automatic-ripping-machine:latest
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/create?fromImage=automaticrippingmachine%2Fautomatic-ripping-machine&tag=latest": dial unix /var/run/docker.sock: connect: permission denied
```

To fix the error, I added the `arm` user to the `docker` group:

```
$ sudo usermod -aG docker arm
```

In addition, there was a minor error at the end of the script. When running the [final line](https://github.com/automatic-ripping-machine/automatic-ripping-machine/blob/cdfdf70aa9def40d4d7d677c47bb8dada6a0888b/scripts/installers/docker-setup.sh#L94), an error is generated, and the path is not printed:

```
./docker-setup.sh: line 94: /home/arm: Is a directory
Installation complete. A template command to run the ARM container is located in:  
```

This is because `~arm` is a reference to a directory, not a path string:

```
$ echo $(~arm)
-bash: /home/arm: Is a directory
```

This can be fixed by using `readlink` instead:

```
$ echo $(readlink -f ~arm)
/home/arm
```

A fix would be to replace line 94 with:

```
echo -e "${RED}Installation complete. A template command to run the ARM container is located in: $(readlink -f ~arm) ${NC}"
```

I hope this helps others who encounter these errors.

**Comment by shitwolfymakes** [comment]
Go ahead and open a PR with the fix for that, so we can credit you!
